daemon/daemon_unix.go

```diff
@@ -77,7 +77,7 @@ func getMemoryResources(config containertypes.Resources) *specs.Memory {
 		memory.Reservation = &reservation
 	}
 
-	if config.MemorySwap != 0 {
+	if config.MemorySwap > 0 {
```

Comment:
This one has been tricky, IIRC.

Seems -1 (previously unlimited) and 0 (unset) are now treated the same.